src/entity/user/experiences/UserExperience.ts

```diff
@@ -1,64 +1,80 @@
   PrimaryGeneratedColumn,
   TableInheritance,
 } from 'typeorm';
 import type { User } from '../User';
-import type { UserSkill } from '../UserSkill';
-import { ExperienceStatus, UserExperienceType } from './types';
+import { UserExperienceType } from './types';
+import type { Company } from '../../Company';
+import { LocationType } from '@dailydotdev/schema';
+import { listAllProtoEnumValues } from '../../../common';
+import type { DatasetLocation } from '../../dataset/DatasetLocation';
+
+const defaultLocationTypes = listAllProtoEnumValues(LocationType);
 
 @Entity()
 @TableInheritance({ column: { type: 'text', name: 'type' } })
 export class UserExperience {
-  @PrimaryGeneratedColumn('uuid')
+  @PrimaryGeneratedColumn('uuid', {
+    primaryKeyConstraintName: 'PK_user_experience_id',
+  })
   id: string;
 
   @Column()
+  @Index('IDX_user_experience_userId')
   userId: string;
 
-  @ManyToOne('User', (user: User) => user.experiences, {
-    onDelete: 'CASCADE',
-  })
-  @JoinColumn({ name: 'userId' })
+  @ManyToOne('User', { lazy: true, onDelete: 'CASCADE' })
+  @JoinColumn({ foreignKeyConstraintName: 'FK_user_experience_user_userId' })
   user: Promise<User>;
 
+  @Column()
+  companyId: string;
+
+  @ManyToOne('Company', { lazy: true, onDelete: 'CASCADE' })
+  @JoinColumn({
+    foreignKeyConstraintName: 'FK_user_experience_company_companyId',
+  })
+  company: Promise<Company>;
+
   @Column({ type: 'text' })
   title: string;
 
-  @Column({ type: 'text', default: '' })
-  description: string;
+  @Column({ type: 'text', nullable: true })
+  subtitle: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  description: string | null;
 
   @Column({ type: 'timestamp' })
-  startDate: Date;
+  startedAt: Date;
 
   @Column({ type: 'timestamp', nullable: true })
-  endDate: Date;
+  endedAt: Date | null;
 
-  @Column({
-    type: 'text',
-    nullable: false,
-  })
+  @Column({ type: 'text', nullable: false })
+  @Index('IDX_user_experience_type')
   type: UserExperienceType;
 
-  @Column({
-    type: 'text',
-    default: ExperienceStatus.Draft,
-  })
-  status: ExperienceStatus;
+  @Column()
+  locationId: string;
 
-  @Column({ type: 'jsonb', default: {} })
-  flags: Record<string, unknown>;
+  @ManyToOne('DatasetLocation', { lazy: true, onDelete: 'SET NULL' })
+  @JoinColumn({
+    name: 'locationId',
+    foreignKeyConstraintName: 'FK_user_experience_dataset_location_locationId',
+  })
+  location: Promise<DatasetLocation>;
 
-  @ManyToMany('UserSkill', (skill: UserSkill) => skill.experiences)
-  @JoinTable({
-    name: 'user_experience_skills',
-    joinColumn: { name: 'experienceId', referencedColumnName: 'id' },
-    inverseJoinColumn: { name: 'skillSlug', referencedColumnName: 'slug' },
+  @Column({
+    type: 'integer',
+    array: true,
+    comment: 'LocationType from protobuf schema',
+    default: defaultLocationTypes,
   })
-  skills: Promise<UserSkill[]>;
+  locationType: Array<LocationType> = defaultLocationTypes;
```

Comment:
You don't need array here, unless you want an experience to have multiple location types?